#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Add the elf entry point definition to those files that are exported.
# DP: Related bugs: None, though there could be possible unknown problems, we never want the
#                   generic version to be used.
# DP: Author: Carlos O'Donell <carlos@baldric.uwo.ca>
# DP: Upstream status: Not submitted
# DP: Status Details: Will be submitted upstream as part of another patch.
# DP: Date: 2004-01-01

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
#
# 2004-01-01  Carlos O'Donell  <carlos@baldric.uwo.ca>
#
#	* sysdeps/hppa/Dist: Export the entry point.
#
--- glibc/sysdeps/hppa/Dist.orig	2004-01-02 11:43:00.000000000 -0500
+++ glibc/sysdeps/hppa/Dist	2004-01-02 11:43:09.000000000 -0500
@@ -1,2 +1,3 @@
 dl-symaddr.c
 dl-fptr.c
+elf/entry.h
